lib/store_web/live/order_live/show.ex

```diff
@@ -0,0 +1,57 @@
+defmodule StoreWeb.OrderLive.Show do
+  use StoreWeb, :live_view
+
+  import Store.Inventory
+  alias Flop.Schema.Store.Inventory.Order
+  alias Store.Repo
+  alias Store.Inventory
+  alias Store.Uploaders
+  alias Store.Accounts
+
+  @impl true
+  def mount(%{"id" => id}, _session, socket) do
+    {:ok, assign(socket, order: Inventory.get_order!(id))}
+  end
+
+  @impl true
+  def handle_params(%{"id" => id}, _, socket) do
+    {:noreply,
+     socket
+     |> assign(:current_page, :orders)
+     |> assign(:page_title, page_title(socket.assigns.live_action))
+     |> assign(:order, Inventory.get_order!(id) |> Repo.preload(:products))}
+  end
+
+  @impl true
+  def handle_event("canceled", _payload, socket) do
+    order = socket.assigns.order
+    Inventory.update_status(order, %{status: :canceled})
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("draft", _payload, socket) do
+    order = socket.assigns.order
+    current_user = socket.assigns.current_user
+    for product <- order.products do
+      case Inventory.purchase(current_user, product) do
+        {:ok, _product} ->
+          {:noreply,
+           socket
+           |> put_flash(:success, "Product purchased successfully!")
+          }
+        {:error, %Ecto.Changeset{} = changeset} ->
+          {:noreply, assign(socket, :changeset, changeset)}
+      end
+    end
+
+    Repo.delete!(order)
+  end
```

Comment:
I think I'm not understanding this function 🤔
It handles the `"draft"` event, but purchases all the products inside a order and then delete it. I'm quite confused 😅

Reply:
Basically I put every product that was in the ```"ordered order"``` in the ```"draft order"``` , so then it makes no sense to still have the order in "ordered" so i deleted that order. :)

Reply:
Hmm, I see.
I don't agree with that. I think a user should only have the possibility to cancel an ordered order.